cmd/end2endtest/ballot.go

```diff
@@ -13,48 +15,87 @@ import (
 func init() {
 	ops["ballotelection"] = operation{
 		test:        &E2EBallotElection{},
-		description: "ballot election with unique values, maxCount, maxValue, maxTotalCost to test different ballotProtocol configurations",
+		description: "ballot election to test different ballotProtocol voting: ranked, quadratic, range and approval",
 		example:     os.Args[0] + " --operation=ballotelection --votes=1000",
 	}
 }
 
 var _ VochainTest = (*E2EBallotElection)(nil)
 
 type E2EBallotElection struct {
-	e2eElection
+	elections []e2eElection
 }
 
 func (t *E2EBallotElection) Setup(api *apiclient.HTTPclient, c *config) error {
-	t.api = api
-	t.config = c
-
-	p := &models.Process{
-		StartBlock: 0,
-		BlockCount: 100,
-		Status:     models.ProcessStatus_READY,
-		EnvelopeType: &models.EnvelopeType{
-			EncryptedVotes: false,
-			UniqueValues:   true},
-		CensusOrigin: models.CensusOrigin_OFF_CHAIN_TREE_WEIGHTED,
-		VoteOptions: &models.ProcessVoteOptions{
-			MaxCount:     2,
-			MaxValue:     6,
-			MaxTotalCost: 10,
-			CostExponent: 1,
-		},
-		Mode: &models.ProcessMode{
-			AutoStart:     true,
-			Interruptible: true,
-		},
-		MaxCensusSize: uint64(t.config.nvotes),
-	}
-
-	if err := t.setupElectionRaw(p); err != nil {
+	var err error
+	if t.elections, err = newE2EElections(api, c, 4); err != nil {
 		return err
 	}
 
-	log.Debugf("election details: %+v", *t.election)
+	//setup for ranked voting
+	p := newTestProcess()
+	p.VoteOptions = &models.ProcessVoteOptions{
+		MaxCount:     4,
+		MaxValue:     3,
+		MaxTotalCost: 6,
+		CostExponent: 1,
+	}
+
+	if err := t.elections[0].setupElectionRaw(p); err != nil {
+		return err
+	}
+
+	log.Debugf("election details: %+v", *t.elections[0].election)
+	// setup for quadratic voting
+	p2 := newTestProcess()
+	// update uniqueValues to false
+	p2.EnvelopeType.UniqueValues = false
+	p2.VoteOptions = &models.ProcessVoteOptions{
+		MaxCount:     4,
+		MaxValue:     3,
+		MaxTotalCost: 12,
+		CostExponent: 2,
+	}
+
+	if err := t.elections[1].setupElectionRaw(p2); err != nil {
```

Comment:
maybe make p an array so you can reference `p[1]` which would look less confusing

this would also let you iterate and deduplicate, as in
 ```
 // not real code
 for i := 0; i<4; i++ {  t.elections[i].setupElectionRaw(p[i])  }
 ```

Reply:
you can also populate the p array with a `for` loop, deduplicating the 4 lines of newTestProcess()